docs: adr for enabling producing to event bus via settings (#224)
openedx/openedx-events: docs/decisions/0012-producing-to-event-bus-via-settings.rst

.. _handlers example: https://github.com/openedx/edx-platform/blob/27b8d2f68d5dfaf84755e7d7f8dccc97ce3be509/cms/djangoapps/contentstore/signals/handlers.py#L162-L210
.. _edx-event-bus-kafka: https://github.com/openedx/event-bus-kafka
.. _edx-event-bus-redis: https://github.com/openedx/event-bus-redis
.. _topic: https://developer.confluent.io/learn-kafka/apache-kafka/topics/#kafka-topics
.. _stream: https://redis.io/docs/data-types/streams/

This ADR aims to propose a solution for configuring the details, like the topic name, consumer group name, etc. via Django settings as well as pushing events to the event bus without requiring the producing application to write additional handlers.

Decision
********

Create a generic signal handler to push events to the event bus. This handler should be attached to or connected to the signals that are enabled in Django settings. The configuration format will be as shown below:

.. code-block:: python

   # .. setting_name: EVENT_BUS_PRODUCER_CONFIG
   # .. setting_default: {}

>    # .. setting_description: Dictionary with event_types as keys and list of dictonaries for topic related configuration.

Reviewer: [nit] Typo, and suggested rewording.

```suggestion
   # .. setting_description: Dictionary of event_types to lists of dictionaries for topic related configuration.
```

   #    Each topic configuration dictionary contains a flag denoting whether the event will be published to the topic,

>    #    topic/stream name where the event will be pushed to, event_key_field which is and period delimited string path

Reviewer: [nit] typo, hyphenation
```suggestion
   #    topic/stream name where the event will be pushed to, event_key_field which is a period-delimited string path
```
